src/AbstractObjects.php

```diff
@@ -112,77 +114,20 @@ protected function processLanguages( $translations ) {
 	 * @return void
 	 */
 	protected function processTranslations( $translations ) {
-		global $wpdb;
-
-		$terms = [];
-
-		foreach ( array_keys( $translations ) as $name ) {
-			$terms[] = $wpdb->prepare( '(%s, %s)', $name, $name );
-		}
-
-		$terms = array_unique( $terms );
-
-		// Insert terms.
-		if ( ! empty( $terms ) ) {
-			$wpdb->query( "INSERT INTO $wpdb->terms (slug, name) VALUES " . implode( ',', $terms ) ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
-		}
+		$translated_object = false;
 
-		// Get all terms with their term_id.
-		$terms = $wpdb->get_results(
-			sprintf(
-				"SELECT term_id, slug FROM $wpdb->terms WHERE slug IN ( '%s' )",
-				implode( "', '", esc_sql( array_keys( $translations ) ) )
-			)
-		);
-
-		$tts = [];
-
-		foreach ( $terms as $term ) {
-			$tts[] = $wpdb->prepare(
-				'(%d, %s, %s, %d)',
-				$term->term_id,
-				$this->getTranslationTaxonomy(),
-				serialize( $translations[ $term->slug ] ), // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_serialize
-				count( $translations[ $term->slug ] )
-			);
-		}
-
-		$tts = array_unique( $tts );
-
-		// Insert term taxonomy part of terms.
-		if ( ! empty( $tts ) ) {
-			$wpdb->query( "INSERT INTO $wpdb->term_taxonomy (term_id, taxonomy, description, count) VALUES " . implode( ',', $tts ) ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
-		}
-
-		unset( $terms, $tts ); // Free some memory.
-
-		// Get all terms with their term taxonomy id.
-		$terms = $wpdb->get_results(
-			sprintf(
-				"SELECT tt.term_taxonomy_id, t.slug FROM $wpdb->terms AS t
-				INNER JOIN $wpdb->term_taxonomy AS tt ON t.term_id = tt.term_id
-				WHERE tt.taxonomy = '%s'
-				AND t.slug IN ( '%s' )",
-				esc_sql( $this->getTranslationTaxonomy() ),
-				implode( "', '", esc_sql( array_keys( $translations ) ) )
-			)
-		);
-
-		$trs = [];
-
-		if ( is_array( $terms ) ) {
-			foreach ( $terms as $term ) {
-				foreach ( $translations[ $term->slug ] as $object_id ) {
-					$trs[] = sprintf( '(%d, %d)', (int) $object_id, (int) $term->term_taxonomy_id );
-				}
+		foreach ( PLL()->model->translatable_objects as $translatable_object ) {
+			if ( $translatable_object->get_tax_translations() === $this->getTranslationTaxonomy() ) {
+				$translated_object = $translatable_object;
+				break;
```

Comment:
Can't you use `PLL_Translatable_Objects::get()` instead of looping through all items?

Reply:
No: `PLL_Translatable_Objects::get()` expects a type (`post`, `term`), while all we have is [a translation taxonomy name](https://github.com/polylang/wpml-to-polylang/blob/65836435c911c0c8a8c98e1597994a056463c7b8/src/Posts.php#L85) via `AbstractObjects::getTranslationTaxonomy()`.

Reply:
Then I suggest to introduce a new method such as `getObjectType()`.
From my point of view, it is a pity to not use the dedicated API from Polylang.

Reply:
OK why not.
Done in [61dcfbe](https://github.com/polylang/wpml-to-polylang/pull/26/commits/61dcfbe151bc25e8825928886a25187e1264bde4).